### PR TITLE
Code coverage with Jacoco and Codecov

### PR DESCRIPTION
**Pull request by marcgs**
This PR introduces Jacoco with Codecov as documented in a [previous spike](https://github.com/agera-edc/DataSpaceConnector/blob/spike/code-quality-and-test-coverage/docs/codequality/coverage-tools/CODECOV.md).

Jacoco attaches an agent to the JVM that is used to inspect when classes are called and what lines are executed. This adds a test execution overhead that causes tests to be slower (based on our observations from ~8m to ~10m total execution time). As a consequence several race conditions in existing tests surfaced, requiring diverse fixes:

- Atomikos timeout increased to 60s to prevent transaction timeouts in `AtomikosTransactionExtensionTest`
- Increased timeout to 60s for `OkHttpClient` instances in tests
- Add initial polling delay in `HealthCheckServiceImplTest` to prevent race conditions
- Additional fix to prevent running duplicate pipelines when pushing changes to open PRs.

Codecov is unable to report coverage difference as there is no data uploaded for the target branch yet. This will work correctly once merged into the main branch.

### CODE REVIEW

File: extensions/transaction/transaction-atomikos/src/test/java/org/eclipse/dataspaceconnector/transaction/atomikos/AtomikosTransactionExtensionTest.java [hunk 1]
```diff
@@ -48,6 +50,8 @@ void verifyEndToEndTransactions() {
         when(extensionContext.getConfig()).thenReturn(ConfigFactory.fromMap(Map.of(TransactionManagerConfigurationKeys.LOGGING, "false")));
```
Comment (Izzzu) [inline extensions/transaction/transaction-atomikos/src/test/java/org/eclipse/dataspaceconnector/transaction/atomikos/AtomikosTransactionExtensionTest.java:50]: Why getConfig() response needs to be recorded twice?

Comment (Izzzu) [reply extensions/transaction/transaction-atomikos/src/test/java/org/eclipse/dataspaceconnector/transaction/atomikos/AtomikosTransactionExtensionTest.java:50]: Could you perhaps add a note explaining this fix in the PR description?

Comment (marcgs) [reply extensions/transaction/transaction-atomikos/src/test/java/org/eclipse/dataspaceconnector/transaction/atomikos/AtomikosTransactionExtensionTest.java:50]: There are 2 invocations, one with a String argument and one without. This is adding mock behaviour for the second one.

Comment (marcgs) [reply extensions/transaction/transaction-atomikos/src/test/java/org/eclipse/dataspaceconnector/transaction/atomikos/AtomikosTransactionExtensionTest.java:50]: Improved PR description

**Review by cpeeyush — APPROVED** [review]

**Pull request merged**